### PR TITLE
Revamp MPC Client

### PR DESCRIPTION
**Pull request by thurstonsand**
* Rework the MPC client to make it easier to use to send messages to other helpers
* Renamed a few structs and functions to make more sense with the current design

### CODE REVIEW

File: src/net/client/mod.rs [hunk 1]
```diff
@@ -1,54 +1,40 @@
...
-
-    #[error(transparent)]
-    AxumError(#[from] axum::http::Error),
-}
-
-#[async_trait]
-pub trait MpcHandle {
-    async fn execute(&self, command: Command) -> Result<Vec<u8>, MpcClientError>;
+pub struct HttpSendMessagesArgs<'a> {
+    pub query_id: &'a QueryId,
```
Comment (akoshelev) [inline src/net/client/mod.rs:21]: do you need a reference here?

File: src/net/client/mod.rs [hunk 2]
```diff
@@ -86,16 +74,26 @@ impl MpcHttpConnection {
...
+    /// some state of transformation within a protocol. Also includes ['`RecordId`] information and
+    /// [`ChannelId`].
+    /// # Errors
+    /// If the request has illegal arguments, or fails to deliver to helper
+    pub async fn send_messages(
+        &self,
+        args: HttpSendMessagesArgs<'_>,
+    ) -> Result<(), MpcClientError> {
         let uri = self.build_uri(format!(
             "/mul/query-id/{}/step/{}?identity={}",
```
Comment (akoshelev) [inline src/net/client/mod.rs]: given the change, you probably want to change the uri too and remove `mul`?

File: src/net/client/mod.rs [hunk 3]
```diff
@@ -1,54 +1,40 @@
...
+    pub identity: Identity,
+    pub offset: u32,
+    pub data_size: u32,
+    pub messages: Bytes,
 }
 
-pub struct MpcHttpConnection {
+#[allow(clippy::module_name_repetitions)] // follows standard naming convention
+#[derive(Clone)]
+pub struct MpcClient {
```
Comment (akoshelev) [inline src/net/client/mod.rs]: nit: maybe `MpcHelperClient`?

File: src/net/client/mod.rs [hunk 4]
```diff
@@ -74,6 +58,10 @@ impl MpcHttpConnection {
         Ok(Self::new(addr.parse()?))
     }
 
+    pub fn origin(&self) -> String {
```
Comment (akoshelev) [inline src/net/client/mod.rs]: need this?

File: src/net/client/error.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    NetworkConnection(#[from] hyper::Error),
+
+    #[error("request returned {status}: {reason}")]
+    FailedRequest {
+        status: hyper::StatusCode,
+        reason: String,
+    },
+
+    #[error(transparent)]
+    AxumError(#[from] axum::http::Error),
```
Comment (akoshelev) [inline src/net/client/error.rs]:
```suggestion
    HttpError(#[from] hyper::http::Error),
```

File: src/net/client/error.rs [hunk 6]
```diff
@@ -0,0 +1,38 @@
+use crate::net::client::MpcClientError::FailedRequest;
+use thiserror::Error as ThisError;
+
+#[derive(ThisError, Debug)]
+#[allow(clippy::module_name_repetitions)] // follows standard naming convention
+pub enum MpcClientError {
+    #[error(transparent)]
+    InvalidHostAddress(#[from] axum::http::uri::InvalidUri),
```
Comment (akoshelev) [inline src/net/client/error.rs]:
```suggestion
    InvalidUri(#[from] hyper::http::uri::InvalidUri),
```

File: src/net/client/mod.rs (hunk 1 above)
Comment (thurstonsand) [reply src/net/client/mod.rs:21]: ah, the compiler complained that `UniqueStepId` could be passed by ref, so I changed both. but yeah, `QueryId` is `Copy`, so should just pass the value.

File: src/net/client/mod.rs (hunk 2 above)
Comment (thurstonsand) [reply src/net/client/mod.rs]: correct, thanks. good catch. i also changed the server-side to match

File: src/net/client/mod.rs (hunk 3 above)
Comment (thurstonsand) [reply src/net/client/mod.rs]: sounds good to me. i also changed server to `MpcHelperServer`

File: src/net/client/mod.rs (hunk 4 above)
Comment (thurstonsand) [reply src/net/client/mod.rs]: ah, I did need this for peer discovery. i can remove it from this PR and add it there

File: src/net/client/error.rs (hunk 5 above)
Comment (thurstonsand) [reply src/net/client/error.rs]: done

File: src/net/client/error.rs (hunk 6 above)
Comment (thurstonsand) [reply src/net/client/error.rs]: done

**Comment by thurstonsand** [comment]
reopening against main repo

**Pull request closed without merging**